src/ClientData/ProcessData.cpp

```diff
@@ -142,4 +142,15 @@ bool ProcessData::IsModuleLoadedByProcess(const ModuleData* module) const {
   return loaded_module->build_id() == module->build_id();
 }
 
+std::vector<std::pair<std::string, std::string>> ProcessData::GetUniqueModulesPathAndBuildId()
+    const {
+  absl::MutexLock lock(&mutex_);
+  std::set<std::pair<std::string, std::string>> module_keys;
```

Comment:
What about making `ModuleKey` a struct instead of using a `std::pair`? It's used in quite some places and it's hard to remember whether the first element is the module path or the second. It might also be error-prone since both are strings.

Reply:
The reason I was hesitant of doing that is this is something used in sets which means we will need std::hash, absl::Hash, less, equals and then test for all of them. With pair you just get it for free. I am not sure if it is worth it.

Reply:
Hmm, yes. True. Feel free to choose what makes more sense to you.